packages/host/app/routes/application.ts

```diff
@@ -8,6 +9,19 @@ interface Model {
 export default class Application extends Route<Model> {
   @service declare fastboot: { isFastBoot: boolean };
 
+  async beforeModel(transition: any): Promise<void> {
+    // Override the matrix URL for testing
+    if (ENV.environment === 'test' || ENV.environment === 'development') {
+      if (transition.to.queryParams.matrixURL) {
+        ENV.matrixURL = transition.to.queryParams.matrixURL;
```

Comment:
generally we set app state like this in a service. it's a bit awkward to actually set state at runtime in the config environment. the config environment state is normally set during build time.

Reply:
perhaps we should use the matrix service that already exists to set the matrix URL. we can set the matrix URL in the service from here when we see a query param, otherwise the matrix service will fallback to the env config.

Reply:
arg, i was just looking thru the matrix service and i forgot that it lazily loads the matrix deps. calling the matrix service from this point would eagerly load the matrix deps. so nevermind--you are doing this the right way.